Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,193 @@
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first time fetch
+    if last_fetch is None:
+        latest_created_time = dateparser.parse(f'{first_fetch_time} UTC')
+    else:
+        latest_created_time = datetime.fromisoformat(last_fetch)
+    messages_query = generate_search_query(latest_created_time, PERMITTED_FROM_ADDRESSES, PERMITTED_FROM_DOMAINS)
+    messages = client.search(messages_query)
+    messages = messages[:50]
+    mails_fetched = []
+    for uid, message_data in client.fetch(messages, 'RFC822').items():
+        message_bytes = message_data.get(b'RFC822')
+        if not message_bytes:
+            continue
+        email_message = mailparser.parse_from_bytes(message_bytes)
+        # The search query filters emails by day, not by exact date
+        email_message_object = Email(email_message)
+        if email_message_object.date > latest_created_time:
+            mails_fetched.append(email_message_object)
+    if mails_fetched:
+        latest_created_time = max(mails_fetched, key=lambda x: x.date).date
+    incidents = [mail.convert_to_incident() for mail in mails_fetched]
+    next_run = {'last_fetch': latest_created_time.isoformat()}
+    if DELETE_PROCESSED:
+        client.delete_messages(messages)
+    return next_run, incidents
+
+
+def generate_search_query(latest_created_time, permitted_from_addresses, permitted_from_domains):
+    permitted_from_addresses_list = argToList(permitted_from_addresses)
+    permitted_from_domains_list = argToList(permitted_from_domains)
+    messages_query = ''
+    if permitted_from_addresses_list + permitted_from_domains_list:
+        messages_query = OR(from_=permitted_from_addresses_list + permitted_from_domains_list).format()
+        # Removing Parenthesis and quotes
+        messages_query = messages_query.strip('()').replace('"', '')
+        # Creating a list of the OR query words
+    messages_query = messages_query.split()
+    messages_query = messages_query + ['SINCE', latest_created_time]
+    return messages_query
+
+
+def test_module(client: IMAPClient) -> str:
+    client.search()
+    return 'ok'
+
+
+def main():
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    params = demisto.params()
+    mail_server_url = params.get('MailServerURL')
+    port = int(params.get('port'))
+    folder = params.get('folder')
+    username = demisto.params().get('credentials').get('identifier')
+    password = demisto.params().get('credentials').get('password')
+    verify_ssl = not params.get("insecure", False)
+
+    first_fetch_time = demisto.params().get('fetch_time', '3 days').strip()
+
+    LOG(f'Command being called is {demisto.command()}')
+    try:
+        with IMAPClient(mail_server_url, ssl=verify_ssl, timeout=20, port=port) as client:
```

Comment:
why is `timeout` set to `20` ?

Reply:
Some reasonable time, if this is not specified there is no timeout at all